hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataSetLockManager.java

```diff
@@ -94,6 +94,8 @@ private String generateLockName(LockLevel level, String... resources) {
             + resources[0] + "volume lock :" + resources[1]);
       }
       return resources[0] + resources[1];
+    } else if (resources.length == 3 && level == LockLevel.DIR) {
+      return resources[0] + resources[1] + resources[2];
```

Comment:
We need check if resource index from 0 to 2 is null before return.

Reply:
Thanks a lot sir, have fixed.